pkg/types/validation/installconfig.go

```diff
@@ -462,22 +463,25 @@ func validateProxy(p *types.Proxy, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Required(fldPath, "must include httpProxy or httpsProxy"))
 	}
 	if p.HTTPProxy != "" {
-		if err := validate.URI(p.HTTPProxy); err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("httpProxy"), p.HTTPProxy, err.Error()))
+		if err := validateURI(p.HTTPProxy, fldPath.Child("httpProxy"), []string{"http"}); err != nil {
+			allErrs = append(allErrs, err...)
 		}
```

Comment:
The `if` is not necessary.
```suggestion
		allErrs = append(allErrs, validateURI(p.HTTPProxy, fldPath.Child("httpProxy"), []string{"http"})...)
```

Reply:
validateURI returns nil if there are no errors and it gets appended to allErrs. This causes https://github.com/openshift/installer/blob/e0761e06048b2f80176f2b0573453b7ca912663e/pkg/asset/installconfig/installconfig.go#L149 to be true because err != nil turns false when the array has value. I'm guessing that is why the if clause was present before this change too. Let me know if there is a better solution.

Reply:
The difference between what you have here and the code linked is the `ToAggregate` call, which converts the slice of errors into a single error. With the code that you have, if `validateURI` returns `nil`, then nothing will be appended.

Reply:
https://play.golang.org/p/4LRc3kSqgzG

Reply:
Yeah got it, thanks! I'll return an empty list just like the other functions instead.

Reply:
Keep it returning `nil`. There is no need to create an empty list.